tools/javac_worker/src/build/please/compile/JavaCompiler.java

```diff
@@ -49,6 +49,13 @@ public void run() {
         }
     }
 
+    /**
+     * Overriding this allows customising the threadpool that is used.
+     */
+    protected ExecutorService createExecutor() {
+        return Executors.newCachedThreadPool();
```

Comment:
A `cachedThreadPool` is a bit problematic under load as it doesn't have a task queue and will create threads as more tasks need to be submitted to it. `fixedThreadPool` does.